ExampleMod/Content/Items/ExampleResearchPresent.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Terraria;
+using Terraria.GameContent.Creative;
+using Terraria.ID;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Content.Items
+{
+	public class ExampleResearchPresent : ModItem
+	{
+
+		public override void SetStaticDefaults() {
+			DisplayName.SetDefault("Example Present");
+			Tooltip.SetDefault("Contains a random accessory! Try researching it for infinite of it!");
+			CreativeItemSacrificesCatalog.Instance.SacrificeCountNeededByItemId[Type] = ItemLoader.ItemCount;
+		}
+
+		public override void SetDefaults() {
+			Item.CloneDefaults(ItemID.GoodieBag);
+		}
+
+		/*this allows for the present to be researched even when you already have infinite of them*/
+        public override bool? CanResearch() {
+			return true;
+        }
+
+        public override void OnResearched(bool fullyResearched) {
+			/*If you managed to be lucky enough to get all presents, it will give you all
+			 accessories*/
+			if (fullyResearched) {
+				for (int i = 1; i < ItemLoader.ItemCount; i++) {
+					Item tr = new Item(i);
+                    if (tr.accessory) {
+						CreativeUI.ResearchItem(tr);
+					}
+                }
+				Main.NewText("You got all accessories!");
+				return;
+			} else {
+				int j = this.Item.stack;
```

Comment:
It's not clear to me exactly what the intended behaviour is here. Needs a comment.
From what I can see, if you research multiple at once, they all get consumed, but you might run out of rolls on the item table (fixed 1000) before that.
If you run out of rolls, you may have actually researched some accessories, but the text will still say "no new accessory was inside"
The `ResearchItem` call doesn't check if it's already been fully researched either...

Reply:
I changed ResearchItem to have a out boolean alreadyResearched, that will indicate if something was already researched before being sent out. only place that will not have it is the original Sacrifice function.